src/adapters/plain.ts

```diff
@@ -0,0 +1,132 @@
+import { IncomingMessage as NodeIncomingMessage } from "unenv/runtime/node/http/_request";
+import { ServerResponse as NodeServerResponse } from "unenv/runtime/node/http/_response";
+import type { App } from "../app";
+import type { HTTPMethod } from "../types";
+import { createError, isError, sendError } from "../error";
+import { H3Event, createEvent } from "../event";
+import { splitCookiesString } from "../utils";
+
+export interface PlainRequest {
+  _eventOverrides?: Partial<H3Event>;
+  context?: Record<string, unknown>;
+
+  method: string;
+  path: string;
+  headers: HeadersInit;
+  body?: null | BodyInit;
+}
+
+export interface PlainResponse {
+  status: number;
+  statusText: string;
+  headers: [string, string][];
+  body?: unknown;
+}
+
+export type PlainHandler = (request: PlainRequest) => Promise<PlainResponse>;
+
+/** @experimental */
+export function toPlainHandler(app: App) {
+  const handler: PlainHandler = (request) => {
+    return _handlePlainRequest(app, request);
+  };
+  return handler;
+}
+
+// --- Internal ---
+
+export async function _handlePlainRequest(app: App, request: PlainRequest) {
+  // Normalize request
+  const path = request.path;
+  const method = (request.method || "GET").toUpperCase() as HTTPMethod;
+  const headers = new Headers(request.headers);
+
+  // Shim for Node.js request and response objects
+  // TODO: Remove in next major version
```

Comment:
What does this actually mean? I did not find any other information regarding this comment.
Does this mean that internal representation as `IncomingMessage/ServerResponse` will change at some point? If yes, is there discussions somewhere on the subject? If no, what was the idea?

For the context, I'm [experimenting on generic handlers](https://github.com/hattipjs/hattip/discussions/45#discussioncomment-9342511) (and middlewares), and currently listing how each popular server implements handlers and middlewares.

Reply:
Some more context on a dedicated repo https://github.com/magne4000/universal-handler